tools/trimmomatic.cwl

```diff
@@ -1,57 +1,97 @@
 #!/usr/bin/env cwl-runner
 
-cwlVersion: "cwl:draft-3"
-
+cwlVersion: v1.0
 class: CommandLineTool
 
+hints:
+- class: EnvVarRequirement
+  envDef:
+   CLASSPATH: /usr/share/java/trimmomatic.jar
+- class: DockerRequirement
+  dockerImageId: dukegcb/trimmomatic
+
 requirements:
-  - $import: trimmomatic-docker.yml
-  - class: InlineJavascriptRequirement
-  - class: ShellCommandRequirement
+#- $import: trimmomatic-docker.yml
+- class: InlineJavascriptRequirement
+- class: ShellCommandRequirement
 
 inputs:
-  - id: java_opts
-    type:
-      - 'null'
-      - string
-    description: "JVM arguments should be a quoted, space separated list (e.g. \"-Xms128m -Xmx512m\")"
-    inputBinding:
-      position: 1
-      shellQuote: false
-  - id: trimmomatic_jar_path
+  phred:
     type: string
+    default: '64'
     inputBinding:
-      position: 2
-      prefix: "-jar"
-  - id: end_mode
-    type: string
-    description: |
-      SE|PE
-      Single End (SE) or Paired End (PE) mode
+      prefix: -phred
+      separate: false
+      position: 4
+    doc: |
+      "33"|"64"
+      -phred33 ("33") or -phred64 ("64") specifies the base quality encoding. Default: -phred64
+  tophred64:
+    type: boolean?
     inputBinding:
-      position: 3
-  - id: nthreads
+      position: 12
+      prefix: TOPHRED64
+      separate: false
+    doc: This (re)encodes the quality part of the FASTQ file to base 64.
+  headcrop:
+    type: int?
+    inputBinding:
+      position: 13
+      prefix: 'HEADCROP:'
+      separate: false
+    doc: |
+      <length>
+      Removes the specified number of bases, regardless of quality, from the beginning of the read.
+      <length>: The number of bases to keep, from the start of the read.
+  tophred33:
+    type: boolean?
+    inputBinding:
+      position: 12
+      prefix: TOPHRED33
+      separate: false
+    doc: This (re)encodes the quality part of the FASTQ file to base 33.
+  nthreads:
     type: int
```

Comment:
This should be in `argument` instead of `inputs` and the value set to `$(runtime.cores)`